File: go/vehicle/removecron.go
package vehicle

import (
	"fmt"
)

// Script to clean up crontabs created for Tesla Climate, Charge Check, and
// Software Update.  Should be set to run in the middle of the day as all
// the crontabs are set for evening or early morning.
func RemoveTeslaCron() {
	values := [2]string{"m3", "mx"}

	for _, val := range values {
		DeleteCronTab(fmt.Sprintf("cd /home/pi/tesla/go && /usr/bin/timeout -k 60 300 go run main.go -%s >> "+
			"/home/pi/tesla/go/cron.log 2>&1", GetInLineSub("start", val, "precondition")))
		DeleteCronTab(fmt.Sprintf("cd /home/pi/tesla/go && /usr/bin/timeout -k 60 300 go run main.go -%s >> "+
			"/home/pi/tesla/go/cron.log 2>&1", GetInLineSub("stop", val, "precondition")))

		DeleteCronTab(fmt.Sprintf("cd /home/pi/tesla/go && /usr/bin/timeout -k 60 300 go run main.go -%s >> "+
			"/home/pi/tesla/go/cron.log 2>&1", GetInLineSub("check", val, "charge")))

		DeleteCronTab(fmt.Sprintf("cd /home/pi/tesla/go && /usr/bin/timeout -k 60 300 go run main.go -%s >> "+
			"/home/pi/tesla/go/cron.log 2>&1", GetInLineSub("schedule", val, "softwareupdate")))
	}
}

File: python/vehicle/removecron.py
from crontab import CronTab


##
# Script to clean up crontabs created for Tesla Climate and Charge Check.  
# Should be set to run in the middle of the day as all the crontabs are 
# evening or early morning.
#
# author: mjhwa@yahoo.com
##
def main():
  cron = CronTab(user='pi')
  job = cron.find_command('/usr/bin/timeout -k 60 300 python -u /home/pi/tesla/python/vehicle/climate.py --start=m3 >> /home/pi/tesla/python/cron.log 2>&1')
  cron.remove(job)
  cron.write()
  job = cron.find_command('/usr/bin/timeout -k 60 300 python -u /home/pi/tesla/python/vehicle/climate.py --stop=m3 >> /home/pi/tesla/python/cron.log 2>&1')
  cron.remove(job)
  cron.write()

  job = cron.find_command('/usr/bin/timeout -k 60 300 python -u /home/pi/tesla/python/vehicle/climate.py --start=mx >> /home/pi/tesla/python/cron.log 2>&1')
  cron.remove(job)
  cron.write()
  job = cron.find_command('/usr/bin/timeout -k 60 300 python -u /home/pi/tesla/python/vehicle/climate.py --stop=mx >> /home/pi/tesla/python/cron.log 2>&1')
  cron.remove(job)
  cron.write()

  job = cron.find_command('/usr/bin/timeout -k 60 300 python -u /home/pi/tesla/python/vehicle/charge.py --check=m3 >> /home/pi/tesla/python/cron.log 2>&1')
  cron.remove(job)
  cron.write()
  job = cron.find_command('/usr/bin/timeout -k 60 300 python -u /home/pi/tesla/python/vehicle/charge.py --check=mx >> /home/pi/tesla/python/cron.log 2>&1')
  cron.remove(job)
  cron.write()

if __name__ == "__main__":
  main()
